Core/clim-core/presentations/presentation-type-functions.lisp

```diff
@@ -104,17 +103,16 @@ otherwise return false."
   'expression)
 
 (flet ((get-ptype-from-class-of (object)
-         (let* ((name (class-name (class-of object)))
-                (ptype-entry (gethash name *presentation-type-table*)))
-           (unless ptype-entry
-             (return-from get-ptype-from-class-of nil))
-           ;; Does the type have required parameters?
-           ;; If so, we can't use it...
-           (let ((parameter-ll (parameters-lambda-list ptype-entry)))
-             (values name
-                     (if (eq (car parameter-ll) '&whole)
-                         (cddr parameter-ll)
-                         parameter-ll))))))
+         (let ((name (class-name (class-of object))))
+           (if-let ((ptype (find-presentation-type name nil)))
+             ;; Does the type have required parameters?
+             ;; If so, we can't use it...
+             (let ((parameter-ll (parameters-lambda-list ptype)))
+               (values name
+                       (if (eq (car parameter-ll) '&whole)
+                           (cddr parameter-ll)
+                           parameter-ll)))
+             (return-from get-ptype-from-class-of nil)))))
```

Comment:
fixed

Reply:
I may be using things incorrectly, but the code appears unchanged. (And the commit "presentations: move the function PRESENTATION-TYPE-OF" must be adjusted as well)

Reply:
OK, I see, there is a new commit at the end. Could you ammend the respective commits?

Reply:
this function is moved to presentation-types.lisp later, so amending it will give me a cascade of merge conflicts. since it is cosmetic, maybe let's keep it as a separate last touch?

Reply:
fixed